app/controllers/sessions_controller.rb

```diff
@@ -7,6 +7,26 @@ def callback
     redirect_back_or(root_url)
   end
 
+  def new
+    if signed_in?
+      redirect_to root_url
+    end
+  end
+
+  def create
+   user = User.find_by_email(params[:session][:email].downcase)
+   if user && user.authenticate(params[:session][:password])
+     first_time_sign_in user
+   else
+     flash.now[:error] = 'Invalid email/password combination'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.